index.ts

```diff
@@ -1,25 +1,20 @@
 'use strict';
 
-Object.defineProperty(exports, "__esModule", { value: true });
+import picomatch, { PicomatchOptions } from 'picomatch';
+import normalizePath from 'normalize-path';
 
-const picomatch = require('picomatch');
-const normalizePath = require('normalize-path');
 
-/**
- * @typedef {(testString: string) => boolean} AnymatchFn
- * @typedef {string|RegExp|AnymatchFn} AnymatchPattern
- * @typedef {AnymatchPattern|AnymatchPattern[]} AnymatchMatcher
- */
```

Comment:
Shall we keep these jsdoc comments instead of turning this file into typescript? TS supports using jsdoc as typings so we don't need to convert to benefit from typing.

Reply:
jsdoc needs to be maintained, also index.d.ts. If you forget to update typings you risk issues like this: https://github.com/micromatch/anymatch/issues/40

Also typescript  with **noImplicitAny** does a strict typing check assuring all types are correctly defined.

There are many other advantages of using typescript, like:
- having multiple targets as output, supporting both commonjs and ecmascipt modules
- using latest ecmascript features and targeting es5 keeping compatibility with nodejs versions.

Reply:
I know the tradeoffs. I suppose I framed the reversion as a question, but I actually meant it as "this won't land without undoing the TS change".

Reply:
Ah you mean you want keep the code in typescript, but use jsdoc instead of using declaring type.
Sure it can be done, but I don't see the advantage apart from reverting 3 lines change.
Typescript syntax is much more flexible and it has no much sense to keep jsodoc in a typescript project, anyway let me know what you prefer, if you want I'll revert it.

Reply:
We subscribe to https://dev.to/thepassle/using-typescript-without-compilation-3ko4 - so please put the jsdoc typings back.